### PR TITLE
Document view

### PR DESCRIPTION
**Pull request by markuswich**
Added sortable document table to the home screen

### CODE REVIEW

File: documentdata.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        self.createdAt = createdAt
+
+    def asDict(self):
+        return {
+            "name" : self.name,
+            "path" : self.path,
+            "type" : self.type,
+            "lang" : self.lang,
+            "size" : self.size,
+            "createdAt" : self.createdAt
```
Comment (alexanderbarnhill) [inline documentdata.py:18]: ~~Is this necessary? I would have thought that returning a jsonified object would return exactly this?~~

It is necessary. My mistake

File: documentdata.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    def __init__(self, name, path, type, lang, size, createdAt):
+        self.name = name
+        self.path = path
+        self.type = type
+        self.lang = lang
+        self.size = size
+        self.createdAt = createdAt
+
+    def asDict(self):
```
Comment (alexanderbarnhill) [inline documentdata.py:11]:
```suggestion
    def as_dict(self):
```
No camel casing in python :)

**Comment by markuswich** [comment]
-Fixed the naming.
-I renamed apiUrl back to serverUrl to be in line with Nazmus' code
-Fixed a bug where the timestamp wasn't correctly send to client and thus broke sorting; timestamps also support hours/minutes/seconds now

Do you think i should add runtime type checking for the constructor of DocumentData?

**Comment by alexanderbarnhill** [comment]
> -Fixed the naming.
> -I renamed apiUrl back to serverUrl to be in line with Nazmus' code
> -Fixed a bug where the timestamp wasn't correctly send to client and thus broke sorting; timestamps also support hours/minutes/seconds now
> 
> Do you think i should add runtime type checking for the constructor of DocumentData?

It wouldn't hurt, but we can also do that later if we find it necessary

**Comment by markuswich** [comment]
I just noticed I should probably add comments to the python code. Do we have a style for that or are we going to use something like doxygen?

**Comment by alexanderbarnhill** [comment]
> I just noticed I should probably add comments to the python code. Do we have a style for that or are we going to use something like doxygen?

I've just been going with the standard pydoc format. But its still early enough that we can change that if you want.

**Pull request merged**